backend/src/main/java/com/bakdata/conquery/models/query/ExternalStateImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.bakdata.conquery.models.query;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.stream.Stream;
+import jakarta.ws.rs.core.Response;
+
+import com.bakdata.conquery.apiv1.execution.ResultAsset;
+import com.bakdata.conquery.io.external.form.ExternalFormBackendApi;
+import com.bakdata.conquery.io.result.ExternalState;
+import com.bakdata.conquery.models.auth.entities.User;
+import com.google.common.collect.MoreCollectors;
+import it.unimi.dsi.fastutil.Pair;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@RequiredArgsConstructor
+public class ExternalStateImpl implements ExternalState {
+	private final CountDownLatch latch;
+
+	@Getter
+	private final ExternalFormBackendApi api;
+
+	@Getter
+	private final User serviceUser;
+
+	/**
+	 * Pairs of external result assets (internal url) and their internal asset builder.
+	 * The internal asset builder generates the asset url with the context of a user request.
+	 */
+	@Setter
+	private List<Pair<ResultAsset, AssetBuilder>> resultsAssetMap = Collections.emptyList();
+
+	@Override
+	public CountDownLatch getExecutingLock() {
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/Condition.html

sowas vlt?

Reply:
Ich verstehe dass so, dass man Condition nutzt, wenn man mehrere States hat, die semantisch auf ein Object abzielen, auf die man synchronisieren will (in dem Bsp. full und empty).  Das Conditions aus einem Lock herausgeboren werden scheint nicht besonders wichtig zu sein.
Wenn man nur einen State hat, wie wir, dann würde ein Object.wait/signal das gleiche bringen und wir müssten uns auch um spurious Wakeups kümmern.

Mit CountdownLatch und Uninterruptibles müssen wir das nicht